assembl/models/attachment.py

```diff
@@ -167,6 +177,36 @@ def __init__(self, *args, **kwargs):
 
     # Should we defer this?
     data = Column(LargeBinary, nullable=False)
+    # Note: use SQLA 1.1 for a better Enum behaviour
+    av_checked = Column(Enum(*AntiVirusStatus.__members__.keys(),
+                             name="anti_virus_status"),
+                        server_default='unchecked')
+
+    def guess_extension(self):
+        extensions = guess_all_extensions(self.mime_type)
+        if extensions:
+            # somewhat random
+            return extensions[0]
+
+    def ensure_virus_checked(self, antivirus=None):
+        "Check if the file has viruses"
+        antivirus = antivirus or get_antivirus()
+        # Lock row to avoid multiple antivirus processes
+        (status,) = self.db.query(File.av_checked).filter_by(id=self.id).with_for_update().first()
+        if status == AntiVirusStatus.failed.unchecked.name:
+            safe = antivirus.check(self.data, self.guess_extension())
```

Comment:
What is the reason to do work about file extension before giving it to the antivirus? Is it really valuable input for the antivirus? (does it not already do this work?)